Add nullable property support to API modeling

fahrigedik opened the pull request:
Introduces IsNullable detection in ReflectionHelper and propagates it to PropertyApiDescriptionModel and TypeScript models. Updates logic to consider nullable properties when determining optionality, improving API schema accuracy for nullable reference types.

### Description

Resolves #24306 (write the related issue number if available)

### How to test it?

[test.md](https://github.com/user-attachments/files/24073847/test.md)

fahrigedik committed:
Add nullable property support to API modeling

Introduces IsNullable detection in ReflectionHelper and propagates it to PropertyApiDescriptionModel and TypeScript models. Updates logic to consider nullable properties when determining optionality, improving API schema accuracy for nullable reference types.

maliming committed:
Remove IsNullable from ReflectionHelper and update usage

maliming commented on npm/ng-packs/packages/schematics/src/utils/model.ts:
If `prop.isRequired` then it is not optional.

https://github.com/abpframework/abp/issues/24306#issuecomment-3621575750

maliming committed:
Merge branch 'rel-10.0' into issue-24306-v2

fahrigedik committed:
Handle nullable properties in model generator

Adds support for marking properties as nullable by appending '| null' to their type in the model generator. Also refines the logic for determining optional properties to better align with 'isRequired' and 'isNullable' flags.

maliming committed:
Improve nullable type detection in API modeling

maliming committed:
Add IsNullable property detection for PropertyInfo

fahrigedik committed:
Update optional property check in model utils

The isOptionalProperty function now only checks if a property is not required, removing the isNullable condition. This simplifies the logic for determining optional properties.

sumeyyeKurtulus commented on npm/ng-packs/packages/schematics/src/utils/model.ts:
```diff
@@ -186,10 +190,7 @@ export function createRefToImportReducerCreator(params: ModelGeneratorParams) {
-    prop.typeSimple.endsWith('?') ||
-    ((prop.typeSimple === 'string' || prop.typeSimple.includes('enum')) && !prop.isRequired)
-  );
+  return !prop.isRequired;
```
Hello @maliming we determine the nullability by adding a `?` mark at the end of the definition. 

According to [this comment](https://github.com/abpframework/abp/issues/24306#issuecomment-3621575750) I guess we need to add a question mark if the property is not required and it is nullable at the same time. If so, we need to update this condition to be `!prop.isRequired && prop.isNullable`

maliming replied:
`IsRequired` is for backend, not Angular code.
`IsNullable` determines whether this property can be null in Angular code.

1. If a property is `IsRequired`, that means it has to exist in the request and not be `null`.
2. If a property is `IsNullable`, that means you can set it to `null` in your Angular app.

JanneHarju commented:
Now I found this what might fix that enum nullable probelm. But is this coming to Abp 9.3?

sumeyyeKurtulus commented:
> Now I found this what might fix that enum nullable probelm. But is this coming to Abp 9.3?

Hello, this change will be effective within the latest patch release for the version 10.
